### PR TITLE
Replace IceUtil::Mutex with std::mutex

### PR DESCRIPTION
**Pull request by pepone**
This is a first pass on replacing IceUtil::Mutex by std::mutex

### CODE REVIEW

File: cpp/include/Ice/FactoryTable.h [hunk 1]
```diff
@@ -5,10 +5,10 @@
 #ifndef ICE_FACTORYTABLE_H
 #define ICE_FACTORYTABLE_H
 
-#include <IceUtil/Mutex.h>
 #include <Ice/UserExceptionFactory.h>
 #include <Ice/ValueFactory.h>
 
+#include <mutex>
```
Comment (bernardnormier) [inline cpp/include/Ice/FactoryTable.h:11]: We should have blank line between include and namespace.

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: cpp/src/Ice/ConnectionI.cpp [hunk 2]
```diff
@@ -397,10 +397,7 @@ Ice::ConnectionI::start(const StartCallbackPtr& callback)
             //
             // Wait for the connection to be validated.
             //
-            while(_state <= StateNotValidated)
-            {
-                wait();
-            }
+            _conditionVariable.wait(lock, [this]{ return _state >= StateNotValidated; });
```
Comment (externl) [inline cpp/src/Ice/ConnectionI.cpp:400]: Is the equality check correct?
```suggestion
            _conditionVariable.wait(lock, [this]{ return _state > StateNotValidated; });
```

File: cpp/src/IceBT/Engine.cpp [hunk 3]
```diff
@@ -1256,7 +1256,7 @@ class BluetoothService : public DBus::Filter, public std::enable_shared_from_thi
         ConnectCallbackPtr _cb;
     };
 
-    IceUtil::Monitor<IceUtil::Mutex> _lock;
+    std::mutex _lock;
```
Comment (externl) [inline cpp/src/IceBT/Engine.cpp:1259]: Can we rename this to `_mutex`? 

Actually staring at the code, you're locking `_mutex` which used to be `_lock`. A bug? Am I missing something?

**Review by externl — APPROVED** [review]
Looks good! See comments.

File: cpp/src/Ice/ConnectionI.cpp (hunk 2 above)
Comment (pepone) [reply cpp/src/Ice/ConnectionI.cpp:400]: Good catch!

**Pull request merged**